core/src/test/scala/org/apache/spark/JobCancellationSuite.scala

```diff
@@ -712,6 +713,140 @@ class JobCancellationSuite extends SparkFunSuite with Matchers with BeforeAndAft
     assert(executionOfInterruptibleCounter.get() < numElements)
  }
 
+  Seq(true, false).foreach { interruptible =>
+
+    val (hint1, hint2) = if (interruptible) {
+      (" not", "")
+    } else {
+      ("", " not")
+    }
+
+    val testName = s"SPARK-50768:$hint1 use TaskContext.createResourceUninterruptibly " +
+      s"would$hint2 cause stream leak on task interruption"
+
+    test(testName) {
+      import org.apache.spark.JobCancellationSuite._
+      withTempDir { dir =>
+
+        // `InterruptionSensitiveInputStream` is designed to easily leak the underlying stream
```

Comment:
It would be great to mention why InterruptionSensitiveInputStream can easily lead to stream leaks and the key conditions, such as creating internal resources during initialize, as highlighted in the PR description.

Reply:
Updated.